### ISSUE TITLE
Improving access to media in data table

### ISSUE THREAD

**Issue by Ig-Rebollo** [kobotoolbox/kpi#2628]
This issue refers to the current audio recording feature and the steps required to access, listening to, and downloading the recordings. 

The following proposals try to improve the user flow with relatively minor changes to make the recordings more accessible.

This issue would be the first step towards implementing larger updates when implementing the NLP audio feature with transcription and translation.

**Comment by Ig-Rebollo** [comment]
@tinok @magicznyleszek @jnm here are my initial proposals to tackle the basic discoverability and accessibility of the audio recordings. Let me know what are your initial thoughts. 

Once we settle this aspect, we can continue to move forward with additional features.

[KoBo Audio feature - basic UX improvements.pdf](https://github.com/kobotoolbox/kpi/files/4476516/KoBo.Audio.feature.-.basic.UX.improvements.pdf)

I am working in parallel on the UX documentation, where I am trying to propose a method we can follow to test each type of project. I'll make this week another issue to share and discuss that document, and we can try to figure out as well what would be the best way to test these mockups (long story short: depends on a number of factors we should discuss at some point)

**Comment by tinok** [comment]
This is great - some quick feedback for the first screen: This makes sense and works intuitively for me. What about non-audio media questions? How would images, videos, and other filetypes (e.g. PDF) be handled in the table?

**Comment by Ig-Rebollo** [comment]
Good point @tinok, I'll give it some thought and create an additional mockup to show other examples. The logic should stay the same, indicate the file type with an icon beside the title of the column, and then buttons beside each file name to either download or preview it. 

The issue would be how to clearly distinguish between the 'see' button at the beginning of the row that displays the Enketo submission, and the 'see' button for, say, a PDF file. Maybe a magnifying glass over a document icon...

**Comment by tinok** [comment]
@Ig-Rebollo I've changed the description of this issue to just the data table, we can open a new issue for improving the media download screen. I'm pasting here the image of the suggested data table changes: 

![image](https://user-images.githubusercontent.com/673511/79353671-5e322800-7f09-11ea-9c15-70d73623ce34.png)

**Comment by Raph33** [comment]
This is great @Ig-Rebollo! Very excited to see this moving. My two cents: 

>  The logic should stay the same, indicate the file type with an icon beside the title of the column, and then buttons beside each file name to either download or preview it.
The issue would be how to clearly distinguish between the 'see' button at the beginning of the row that displays the Enketo submission, and the 'see' button for, say, a PDF file. Maybe a magnifying glass over a document icon...

What if clicking on the file name was the main action to open the file? We would keep the "play" button for audio files because they can be played directly from the table view, but for others there would just be the download button and file name. 

Clicking on file name would open the file in a full screen visualizer, where the audio or video files could be played, and images and PDFs visualized. And the user could also navigate left and right to other files from the same form.

**Comment by Ig-Rebollo** [comment]
@tinok @Raph33 great suggestions thanks for the input!

Definitely removing all those buttons and simply displaying a link might be a better -and simpler- option, I like it! Another -more complicated- option would be to have two levels in the cell, displaying on top the name of the file and bellow buttons with text specifying what it is. (i.e. 'download' or 'preview'). This would probably be the clearest alternative and eliminate any confusions, but it might not be worth it if the simpler link version works well with users. Maybe we can create those two mockups and test them.

**Comment by Ig-Rebollo** [comment]
@tinok @jnm @magicznyleszek 

Here are some new versions for the table view, specifically looking at how the files would be dealt with. As you may notice looking at the wireframes, there are a few possible directions:

**The simple approach**: the audio could simply have a download button and a link, just like the other file types, that takes you to a different window where you can listen to it, pause it, etc. as @Raph33 proposed. 

**The intermediate option**: We include the play button on the table view for a quick play of the recording, but to see more audio options (i.e. pause, skip forward/backward 30s, etc.) the user would still have to click on the file link.

**The advanced solution**: Almost everything can be done on the table view. The play button would activate a small pop-up window fixed on the bottom-right corner of the table view, that will allow you to control what to play, pause, etc. but also close the pop-up (which would stop the audio) and go back to the row where the audio is playing (in the other cases, if the user plays the recording but scrolls down, they may loose track of where the audio was playing from). Going even one step further, this pop-up window approach could also be used to preview videos and images. In any case, the link would be maintained, in case the user wants to open several files in different tabs, or have a bigger view or more controls.

Below the mockups (I can produce higher definition ones once we decide which direction are we taking)

![KoBo table - files](https://user-images.githubusercontent.com/22472326/79485314-01a33b80-7fe3-11ea-810b-172d888acaf3.png)
![Audio feature - Table view](https://user-images.githubusercontent.com/22472326/79485330-07991c80-7fe3-11ea-9b29-91b6acff22ff.png)

**Comment by Raph33** [comment]
Nice @Ig-Rebollo, all options are great. 

I would love to see the advanced solution implemented (though it's easy to say for someone who isn't developing). In fact, at first I was hoping to be able to preview all media (including photos and videos as small thumbnail) directly in the table, but increase row heights to accommodate that may make the table too messy (with different row heights for forms that have photo/video attachments and forms that don't). 

My main question is: what is the point of the small pop-up window, instead of a larger visualization component? Is it to continue browsing through the data while visualizing/listening to the attachment? If so, what will happen when the user clicks on a new tab (e.g. the "Form" tab at the top, or "Reports" on the left), or even opens an Enketo submission? The pop-up would close automatically, correct?

Unless there is a specific need for visualizing/listening to an attachment _while_ browsing data (which is a need we haven't encountered), I would favor a larger visualization component. 

And to be clear, while I think the advanced solution would be fantastic, even the simple approach will be great.

**Comment by Ig-Rebollo** [comment]
Hi @Raph33 thanks for your comments! Yes indeed, the purpose of having a small pop-up in the corner is to keep browsing data while you listen. Essentially, allowing users to manage everything without having to open many tabs. 

But I completely agree this is not something worth prioritizing if there is no need for it. I would be happy to go with the intermediate option for now, and see how it tests with some users.
